src/domain/profile/ProfileProvider.tsx

```diff
@@ -0,0 +1,53 @@
+import React from 'react';
+
+import LoadingSpinner from '../../common/components/spinner/LoadingSpinner';
+import { ProfileContext } from './ProfileContext';
+import useProfile from './hooks/useProfile';
+import { MyProfile } from './types/ProfileQueryTypes';
+import { useIsLoginReady } from '../auth/useIsLoginReady';
+
+export default function ProfileProvider({
+  children,
+}: {
+  children: React.ReactNode;
+}) {
+  const [profile, setProfileToContext] = React.useState<MyProfile | null>(null);
+
+  const isLoginReady = useIsLoginReady();
+
+  const [fetchProfile, { data, loading, refetch, called }] = useProfile({
+    setProfileToContext: setProfileToContext,
+  });
+
+  const isLoading = loading && !called;
+
+  const clearProfile = () => setProfileToContext(null);
+
+  const refetchProfile = () => {
+    (async () => await refetch())();
+    setProfileToContext(data?.myProfile || null);
+  };
```

Comment:
Hmm... doesn't it now work so that if you call `refetchProfile` it runs the first line with the function call `(async () => await refetch())();` but as the function is asynchronous it won't wait for the function call to return anything but will just immediately run the following line `setProfileToContext(data?.myProfile || null);`? Is your intention here to wait for the refetching to happen first and after that has happened, then to set the refetched myProfile data into the context?

Reply:
The refetch is an async function, so it needs to be waited, because it returns a promise. The refetchProfile however, is not an async function and I don't even want it to be one. Using a then-method to attach the callback function could look better here, but as far as I know, the result should be exaxctly the same.

I can change this to 
```
    refetch().then(() => {
      setProfileToContext(data?.myProfile || null);
    });
```

Reply:
An IIFE (Immediately Invoked Function Expression) is a JavaScript function that runs as soon as it is defined:
https://developer.mozilla.org/en-US/docs/Glossary/IIFE

Reply:
It really seems that it is not waiting for the response
https://stackoverflow.com/a/54758455.

Reply:
So, the then-block should be better and it's now in use. Good call!

Reply:
This describes this better https://stackoverflow.com/questions/58027037/unexpected-behavior-async-await-when-required-with-iife